Scripts/Runtime/Data/Collections/UnsafeTypedStream.cs

```diff
@@ -34,9 +34,14 @@ public unsafe struct UnsafeTypedStream<T> : INativeDisposable
         [BurstCompatible]
         internal struct BufferInfo
         {
+            public static readonly int SIZE = UnsafeUtility.SizeOf<BufferInfo>();
+            public static readonly int ALIGNMENT = UnsafeUtility.AlignOf<BufferInfo>();
```

Comment:
`BufferInfo` was an outside type at first but we wanted to bring it inside because then we'd have to qualify the name for any other collections that also have `BufferInfo`s. It would become `UnsafeTypedStreamBufferInfo` etc so as not to conflict. 

The addition of two ints to the global memory of the app per variant I think is ok for the niceties in developing with it being an inner type. 

Valid point though.

Reply:
Makes sense. I'll pay the int costs for the better names.